File: 프로그래머스/2/12946. 하노이의 탑/하노이의 탑.py
def solution(n):
    def hanoi(n, start, temp, end):
        if n == 1:
            return [[start, end]]
        return hanoi(n-1, start, end, temp) + [[start, end]] + hanoi(n-1, temp, start, end)
    
    return hanoi(n, 1, 2, 3)

File: 프로그래머스/2/12946. 하노이의 탑/하노이의 탑.js
function solution(n) {
    const hanoi = (n, from, temp, to) => {
        if (n === 1) return [[from, to]];
        return [...hanoi(n-1, from, to, temp), [from, to], ...hanoi(n-1, temp, from, to)];
    }
    
    return hanoi(n, 1, 2, 3);
}